app/worker/http.go

```diff
@@ -39,11 +40,25 @@ var makeHTTPRequest = func(req *http.Request, timeout time.Duration) (*http.Resp
 func processHTTPTask(httpTask *model.HTTPTask, job *model.Job,
 	workflow *model.Workflow, l *log.Logger) (*model.TaskResult, error) {
 	uri := processJobString(httpTask.URI, workflow, job)
-	payloadString := processJobString(httpTask.Body, workflow, job)
-	payloadString = maybeExecuteGoTemplate(
-		payloadString,
-		job.InputParameters.Map(),
-	)
+
+	var payloadString string
+	if len(httpTask.FormData) > 0 {
```

Comment:
just a nitpick, and perhaps something to follow up on, maybe we could introduce some field in the httpTask to determine the payload type. like a flag, so we check it instead of the above.

Reply:
Good idea, not sure how to implement it in a clean way, though. Shall we introduce a new field (payload type) as a sort of enum, and set a default? I'm not sure it makes it more obvious and readable than having a fallback implementation (form data, if available, otherwise body). Something we can definitely follow-up.